### DISCUSSION TITLE
Microsoft should invest more in NSWag and Swashbuckle

### DISCUSSION THREAD

**Discussion by lindeberg** [dotnet/aspnetcore#54316] [General]
Microsoft should invest more in NSWag and Swashbuckle? 

Generating OpenAPI schemas and clients is such an essential, powerful, and widely requested tools.

Microsoft is known to invest in the entire ecosystem, it's awesome!

In Microsoft's docs, NSwag and Swashbuckle are the official recommendation.

These packages are mainly maintained by one person each. While the surrounding tech is making rapid advancements, and there are hundreds of feature request and bug reports.

Please, invest more in these essential tools!

**Comment by KennethHoff** [comment]
There have been talks/rumors of a first party json schema generator. Is there any news on that front, @captainsafia ?

**Comment by captainsafia** [comment]
@KennethHoff Yes, I've been exploring adding built-in support for OpenAPI doc generation in our `Microsoft.AspNetCore.OpenApi` package. The package currently only supports generating `OpenApiOperation`s for a single minimal API endpoint. I'm trying to expand it to cover generating the entire doc for both minimal and MVC. If there's particular scenarios you're really keen to make sure we get working with built-in support, let me know.

WRT to more official info on this, there should be something published soon! Thanks for your pateince on this.

**Reply by kimbell** [reply]
I'm waiting in anticipation for any news on this. 
The current maintainer of Swashbuckle is packing up and moving on to other things. 

https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2759#issuecomment-1936136992

**Reply by martincostello** [reply]
FYI ways of continuing the maintenance of Swashbuckle by the wider community are being explored with the maintainer.
